### PR TITLE
engine: Split subdirectory to store to DB

### PR DESCRIPTION
**Pull request by uran0sH**
If we have a directory A, its subdirectories are B, C. We will store it in the following format: A -> 4, A-B -> B, A-C -> C, B -> 2, C -> 2.

### CODE REVIEW

File: src/server/storage_engine/meta_engine.rs [hunk 1]
```diff
@@ -137,45 +159,44 @@ impl MetaEngine {
...
+            Some(value) => {
+                let mut result = Vec::with_capacity(size as usize);
+                let mut l = usize::from_le_bytes(value.as_slice().try_into().unwrap());
+                let mut total = 0;
+                for item in self.dir_db.db.iterator(IteratorMode::From(
+                    format!("{}-", path).as_bytes(),
+                    rocksdb::Direction::Forward,
+                )) {
+                    if offset > 0 {
+                        offset -= 1;
```
Comment (luanshaotong) [inline src/server/storage_engine/meta_engine.rs:174]: Should add `l -= 1`.

Comment (luanshaotong) [reply src/server/storage_engine/meta_engine.rs:174]: But I think the database trait should implement `seek` operation and move this code in it.

File: src/server/storage_engine/meta_engine.rs [hunk 2]
```diff
@@ -185,25 +206,21 @@ impl MetaEngine {
...
-                    bincode::serialize(&sub_dirs).unwrap(),
-                )?;
+        let ft = match file_type {
+            3 => "d",
+            _ => "f",
+        };
+        match self.dir_db.db.get(parent_dir)? {
+            Some(value) => {
+                let len = usize::from_le_bytes(value.as_slice().try_into().unwrap());
+                self.dir_db.db.put(parent_dir, (len + 1).to_le_bytes())?;
```
Comment (luanshaotong) [inline src/server/storage_engine/meta_engine.rs:216]: Are the names unordered?

Comment (uran0sH) [reply src/server/storage_engine/meta_engine.rs:216]: alphabetical order

File: src/server/storage_engine/meta_engine.rs (hunk 1 above)
Comment (uran0sH) [reply src/server/storage_engine/meta_engine.rs:174]: > But I think the database should implement `seek` operation and move this code in it.

Rocksdb doesn't implement this function

Comment (uran0sH) [reply src/server/storage_engine/meta_engine.rs:174]: > But I think the database should implement `seek` operation and move this code in it.

I don't find that rocksdb implements this function.

Comment (uran0sH) [reply src/server/storage_engine/meta_engine.rs:177]: > Should add `l -= 1`.

Done

**Review by luanshaotong — APPROVED** [review]
LGTM

**Pull request merged**